app/data/schemas/message.js

```diff
@@ -0,0 +1,11 @@
+const Joi = require('joi')
```

Comment:
It's quite difficult to read these all as separate schemas when the ack message is so simple.  I think one schema would make more sense for an acknowledgement, with only those reusable ones pulled out into individual ones.

Reply:
Does this sit in the best position within the repo and would that help with readability?

- if we moved `acknowledgement` under its `acknowledgement/schemas` folder (or perhaps the subfolder isn't needed)
- towards the root (maybe under `data` where it is?) have a `schemas` folder which would house the "components", i.e. those reusable ones?

whatcha think?


also, I am testing this `acknowledgement` schema  -- what should I be looking for?

Reply:
I think it best sits in the acknowledgement module as that's where we've encapsulated all of that behaviour.  The intention of `data` was to manage database models and connectivity.  I think this is a different concern.

In terms of validating the ack object, there's a balancing act as we want to make sure we get what we need to process the acknowledgement, but we don't want to add unnecessary constraints to how those acknowledgements can evolve outside of the service.  Like the other services we don't want to reject something that actually has all we need to process otherwise those payments will be lost.  So only need to ever validate the minimum we need.

So here we just need the invoice number and whether it was acknowledged or not.  If it fails we should get a message, but DAX doesn't always send one so may be undefined.

So I'd say we just need a schema that accepts string for invoice (format will vary from scheme to scheme) and bool for success and date for acknowledged and allow any other field.

I would do that in one simple schema file so it's easy for a reader to understand what we're validating.